.github/workflows/java-cd.yml

```diff
@@ -244,26 +238,50 @@ jobs:
               uses: ./.github/workflows/install-shared-dependencies
               with:
                   os: ${{ matrix.host.OS }}
+                  engine-version: "7.2.5"
                   target: ${{ matrix.host.TARGET }}
                   github-token: ${{ secrets.GITHUB_TOKEN }}
+                  use-cache: false
 
             - name: Install protoc (protobuf)
               uses: arduino/setup-protoc@v3
               with:
                   version: "26.1"
                   repo-token: ${{ secrets.GITHUB_TOKEN }}
 
+            - name: Find available port
```

Comment:
Use `cluster_manager` script and parse its output to get the port where server is running.